src/views/withNavigationFocus.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import hoistStatics from 'hoist-non-react-statics';
+
+export default function withNavigationFocus(Component) {
+  class ComponentWithNavigationFocus extends React.Component {
+    static displayName = `withNavigationFocus(${Component.displayName ||
+      Component.name})`;
+
+    state = {
+      isFocused: false,
+    };
+
+    componentDidMount() {
+      if (this.props.navigation) {
+        this.subs = [
+          this.props.navigation.addListener('didFocus', () =>
+            this.setState({ isFocused: true })
+          ),
+          this.props.navigation.addListener('willBlur', () =>
+            this.setState({ isFocused: false })
+          ),
+        ];
+      } else {
+        console.warn(
```

Comment:
Once we start accepting context from this HOC, I think it would best to have an invariant here instead of warning- things would need to be very broken.